tests/shell/shell_recorder/shell_recorder.sh

```diff
@@ -1,12 +1,24 @@
 #!/bin/bash
 
+getTempDir()
+{
+	if [ -z "$TMPDIR" ]; then
+		TMPDIR="/tmp"
+	else
+		if [[ "$TMPDIR" == */ ]]; then
+			TMPDIR=$(echo $TMPDIR | head -c -2)
+		fi
+	fi
+	echo "$TMPDIR"
+}
+
 FILE=$1
 Mountpoint=
 DBFile=
 Storage=
 MountArgs=
 DiffType=File
-OutFile=$(mktemp "${TMPDIR:-/tmp}/tmp.XXXXXXXXX")
+OutFile=$(mktemp "$(getTempDir)/tmp.XXXXXXXXX")
```

Comment:
Does the command 
```bash
mktemp -t tmp
```
work for your use case? The above command returns a file located at a path that looks something like this: `/var/folders/hx/flbncdhj4fs87095gzxvnj3h0000gn/T/tmp.RrhXNVNZ` on my computer (OS X).